src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -779,7 +779,7 @@ private static void WriteSetColorString(bool foreground, ConsoleColor color)
             // Changing the color involves writing an ANSI character sequence out to the output stream.
             // We only want to do this if we know that sequence will be interpreted by the output.
             // rather than simply displayed visibly.
-            if (Console.IsOutputRedirected)
+            if (!SupportsAnsiColor())
```

Comment:
We wouldn't want to read the environment variable and parse it every time a color was changed.  It'd be better to cache it in a static.